src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -479,38 +475,232 @@ def prepare_or_run_pipeline(
             if settings.pipeline_tags
             else None,
         )
-        execution = pipeline.start()
-        logger.warning(
-            "Steps can take 5-15 minutes to start running "
-            "when using the Sagemaker Orchestrator."
-        )
 
-        # Yield metadata based on the generated execution object
-        yield from self.compute_metadata(
-            execution=execution, settings=settings
-        )
+        # Handle scheduling if specified
+        if deployment.schedule:
+            if settings.synchronous:
+                logger.warning(
+                    "The 'synchronous' setting is ignored for scheduled pipelines since "
+                    "they run independently of the deployment process."
+                )
+
+            events_client = session.boto_session.client("events")
+            rule_name = f"zenml-{deployment.pipeline_configuration.name}"
+
+            # Determine first execution time based on schedule type
+            if deployment.schedule.cron_expression:
+                # AWS EventBridge requires cron expressions in format: cron(0 12 * * ? *)
+                # Strip any "cron(" prefix if it exists
+                cron_exp = deployment.schedule.cron_expression.replace(
+                    "cron(", ""
+                ).replace(")", "")
+                schedule_expr = f"cron({cron_exp})"
+                next_execution = None
+            elif deployment.schedule.interval_second:
+                minutes = max(
+                    1,
+                    int(
+                        deployment.schedule.interval_second.total_seconds()
+                        / 60
+                    ),
+                )
+                schedule_expr = f"rate({minutes} minutes)"
+                next_execution = (
+                    datetime.utcnow() + deployment.schedule.interval_second
+                )
+            elif deployment.schedule.run_once_start_time:
+                # Convert local time to UTC for EventBridge
+                dt = deployment.schedule.run_once_start_time.astimezone(
+                    timezone.utc
+                )
+                schedule_expr = f"cron({dt.minute} {dt.hour} {dt.day} {dt.month} ? {dt.year})"
+                next_execution = deployment.schedule.run_once_start_time
+
+            # Create IAM policy for EventBridge
+            iam_client = session.boto_session.client("iam")
+            role_name = self.config.execution_role.split("/")[
```

Comment:
Isn't the execution role the one that should be assumed by the pipeline during execution, and not used for something around the pipeline execution like this scheduling? This seems more like it should be tied to the role of the service connector if applicable or solved in some other way.